src/silx/gui/plot/LegendSelector.py

```diff
@@ -951,43 +976,27 @@ def _legendSignalHandler(self, ddict):
             ret = dialog.exec()
             if ret:
                 newLegend = dialog.getText()
-                self.renameCurve(oldLegend, newLegend)
+                wasActive = self.plot.getActiveCurve() is curve
+                self.plot.removeItem(curve)
+                curve.setName(newLegend)
+                self.plot.addItem(curve)
```

Comment:
It is still not possible to change the name while the item is in the plot.
This restriction can probably be lifted now, but let's keep it a bit longer.